### PR TITLE
do not panic quanton on external i2c errors

### PR DESCRIPTION
**Pull request by lolorc**
following #1708 comments and #1728

### CODE REVIEW

File: flight/targets/quanton/fw/pios_board.c [hunk 1]
```diff
@@ -624,7 +624,7 @@ void PIOS_Board_Init(void) {
 		}
 
 		if (PIOS_I2C_CheckClear(pios_i2c_usart1_adapter_id) != 0)
-			panic(6);
```
Comment (peabody124) [inline flight/targets/quanton/fw/pios_board.c]: i still don't entirely see this one. Can you confirm that if you have a sensor plugged in but not powered it pulls the bus down? I would think an I2C peripheral would be smart enough not to do that but perhaps not.

**Comment by xdandys** [comment]
Coincidentally I tested this last weekend and can confirm that unpowered sensor pulls the bus down :( At least HMC5983. But it actually make sense, because there are internal protective diodes from I2C pins to positive power supply.

File: flight/targets/quanton/fw/pios_board.c (hunk 1 above)
Comment (lolorc) [inline flight/targets/quanton/fw/pios_board.c]: I just tested again with a attiny as a i2c slave and also with a an external mag (HMC5883 GY-273).
If unpowered they pull the bus down and checkclear fails.

**Comment by peabody124** [comment]
OK thanks for for humoring me and testing this. Will review this later. Should be easy then.

**Comment by ernieift** [comment]
Not tested this, but makes sense. The HMC5883 check would fail without the device. This is quite ok. I think the changes are made for picoc. No one wants the FC in a panic loop when you use an unpowered external display or LED stripe.

File: flight/targets/quanton/fw/pios_board.c (hunk 1 above)
Comment (mlyle) [inline flight/targets/quanton/fw/pios_board.c]: @peabody124 The ESD protection diodes on the parts pull the bus down.

**Comment by mlyle** [comment]
Looks good.

File: flight/targets/quanton/fw/pios_board.c (hunk 1 above)
Comment (theothercliff) [inline flight/targets/quanton/fw/pios_board.c]: I looked at a GPS/mag unit that I have this problem with, and it has pull up resistors on SCL and SDA.  They effectively become pull downs if the device is connected, but unpowered.

**Pull request merged**